include/boost/hana/fwd/set.hpp

```diff
@@ -204,9 +204,63 @@ BOOST_HANA_NAMESPACE_BEGIN
     };
 #endif
 
+    //! Returns the intersection of two sets.
+    //! @relates hana::set
+    //!
+    //! Given two sets `xs` and `ys`, `intersection(xs, ys)` is a new set
+    //! containing exactly those elements that are present both in `xs` and
+    //! in `ys`.
+    //! In other words, the following holds for any object `x`:
+    //! @code
+    //!     x ^in^ intersection(xs, ys) if and only if x ^in^ xs && x ^in^ ys
+    //! @endcode
+    //!
+    //!
+    //! @param xs, ys
+    //! Two sets to intersect.
+    //!
+    //!
+    //! Example
+    //! -------
+    //! @include example/set/intersection.cpp
+#ifdef BOOST_HANA_DOXYGEN_INVOKED
+    constexpr auto intersection = [](auto&& xs, auto&& ys) {
+        return tag-dispatched;
+    };
+#endif
     //! Equivalent to `to<set_tag>`; provided for convenience.
     //! @relates hana::set
     constexpr auto to_set = to<set_tag>;
 BOOST_HANA_NAMESPACE_END
 
+    //! Returns the difference of two sets.
```

Comment:
If you're talking about lines removed in include/boost/hana/fwd/difference.hpp, then I think it makes sense to remove them, as we have different implementation for set and map. And we have different explanation/docs for that, too.